### PR TITLE
Extract and move Image Editor to blocks/

### LINKED ISSUE #8372: Image Editor: Move Component Out of Media Modal for better reuse
Let's work to make the component reusable, as this could be used in other areas, like for cropping avatars.  If possible, let's move this to the blocks section, and abstract any leftover media modal hooks.

### PR DESCRIPTION
**Pull request by lamosty**
Fixes #8372. Moves image editor from `client/post-editor/media-modal/image-editor` to `client/blocks/image-editor`.

The approach I've taken is to:

1. Move the image editor to `blocks/` because it accesses Redux state and is coupled to Calypso;
2. Name the folder `image-editor` as it consists of multiple parts such as `image-editor-canvas`, `image-editor-toolbar`, `image-editor-buttons`. I know that @retrofox had a bit different approach in [8372](https://github.com/Automattic/wp-calypso/issues/8372#issuecomment-250847189) but I think this is a bit more correct.

The following things have been done as a part of this PR:

1. Refactored names of components;
2. Refactored components to ES6 where possible;
3. Fix classNames;
4. Fix indentations;
5. Fix smaller things such as using `Gridicon` of size `12` instead of `14` which is not supported.k
6. and more...

## Testing Instructions

1. Navigate to a post, open media modal and try to edit an image;
2. Is the image editor the same as on https://wpcalypso.wordpress.com/; Does it behave the same?
3. Navigate to http://calypso.localhost:3000/devdocs/blocks/image-editor. Is the image editor usable in context of devdocs? Is it easily reusable?
4. Read https://github.com/Automattic/wp-calypso/blob/a12002846931093736156aa63b2a7ba4e4ea92ec/client/blocks/image-editor/README.md. Is it detailed enough? No big spelling/grammar mistakes?

## Todo

- [x] Add `image-editor` block component to devdocs
- [x] Add readme

/cc @gwwar @retrofox

### CODE REVIEW

**Comment by gwwar** [comment]
@lamosty do you mind moving the folders over in 3f4cb74 instead of a copy. I'm pretty sure we'll lose history on the files otherwise

**Comment by lamosty** [comment]
> @lamosty do you mind moving the folders over in 3f4cb74 instead of a copy. I'm pretty sure we'll lose history on the files otherwise

Oh, yep, I'll do that. I forgot that `git` can preserve history if moved.

**Comment by retrofox** [comment]
I think the style filename should be `style.scss`.

**Comment by lamosty** [comment]
> I think the style filename should be style.scss.

Yep, WIP. :)

File: client/post-editor/media-modal/index.scss [hunk 1]
```diff
@@ -28,6 +28,7 @@
 .editor-media-modal .dialog__content {
 	position: static;
 	color: $gray-dark;
+	height: 100%;
```
Comment (lamosty) [inline client/post-editor/media-modal/index.scss]: I didn't find any regressions in editor media modal. This really helps with image editor styling inside and outside editor media modal (see http://calypso.localhost:3000/devdocs/blocks/image-editor)

**Comment by gwwar** [comment]
Maybe disable the "Done" button on the example blocks page? It throws a js error currently

File: client/blocks/image-editor/image-editor-toolbar.jsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+				label: translate( '3:2' )
+			}
+		];
+
+		return (
+			<PopoverMenu isVisible={ showAspectPopover }
+				onClose={ this.onAspectClose }
+				position="top"
+				context={ popoverContext }
+				className="popover is-dialog-visible"
```
Comment (gwwar) [inline client/blocks/image-editor/image-editor-toolbar.jsx]: I'm getting a eslint warning here for `warning  className should follow CSS namespace guidelines (expected image-editor__ prefix)  wpcalypso/jsx-classname-namespace`. I think we can fix this up by passing through an additional class name here

File: client/blocks/image-editor/index.jsx [hunk 3]
```diff
@@ -25,29 +26,31 @@ import {
...
-		title: React.PropTypes.string,
-		translate: React.PropTypes.func,
-		onImageEditorClose: React.PropTypes.func,
-		onImageEditorCancel: React.PropTypes.func
+		// Component props
+		media: PropTypes.object,
+		siteId: PropTypes.number,
+		onImageEditorClose: PropTypes.func,
+		onImageEditorCancel: PropTypes.func,
+		additionalClasses: PropTypes.string,
```
Comment (gwwar) [inline client/blocks/image-editor/index.jsx]: This works, but why not use the `className` prop directly?

Usage becomes:
```javascript
const classes = classNames( 'image-editor', this.props.className );
```

Comment (lamosty) [inline client/blocks/image-editor/index.jsx]: Hmm, good idea. I saw the `additionalClasses` somewhere but this is much better. Thanks!

File: client/post-editor/media-modal/index.jsx [hunk 4]
```diff
@@ -396,11 +396,18 @@ module.exports = React.createClass( {
...
+				const selectedIndex = this.getDetailSelectedIndex(),
+					media = items ? items[ selectedIndex ] : null;
+
 				content = (
 					<ImageEditor
-						site={ this.props.site }
-						items={ this.props.mediaLibrarySelectedItems }
-						selectedIndex={ this.getDetailSelectedIndex() }
+						site={ site }
+						media={ media }
```
Comment (gwwar) [inline client/post-editor/media-modal/index.jsx]: 👍 Good call to pass this through

File: client/blocks/image-editor/docs/example.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Internal dependencies
+ */
+import ImageEditor from '../';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+function ImageEditorExample( { primarySiteId } ) {
+	const media = {
+		URL: 'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/SIPI_Jelly_Beans_4.1.07.tiff/lossy-page1-256px-SIPI_Jelly_Beans_4.1.07.tiff.jpg'
```
Comment (gwwar) [inline client/blocks/image-editor/docs/example.jsx]: Maybe let's choose an image we host? eg on CloudUp or something from one of our CDNs

**Comment by gwwar** [comment]
I left a few notes, but couldn't find any regressions. Great work @lamosty !

File: client/blocks/image-editor/docs/example.jsx (hunk 5 above)
Comment (lamosty) [inline client/blocks/image-editor/docs/example.jsx]: I wanted to add an image which has CORS enabled. Didn't try CloudUp though. Great suggestion!

**Comment by lamosty** [comment]
Thanks for the reviews!

**Pull request merged**